joern-cli/frontends/kotlin2cpg/src/main/scala/io/joern/kotlin2cpg/types/NameRenderer.scala

```diff
@@ -0,0 +1,225 @@
+  private var anonObjectCounter     = 0
+
+  def descName(desc: DeclarationDescriptor): String = {
+    if (desc.getName.isSpecial) {
+      desc match {
+        case _: ConstructorDescriptor =>
+          Defines.ConstructorMethodName
+        case functionDesc: FunctionDescriptor =>
+          Defines.ClosurePrefix + getAnonDescIndex(desc)
+        case _ =>
+          "object$" + getAnonDescIndex(desc)
+      }
+    } else {
+      desc.getName.getIdentifier
+    }
+  }
+
+  def descFullName(desc: DeclarationDescriptor): Option[String] = {
+    val dealiasedDesc =
+      desc match {
+        case typeAliasDesc: TypeAliasConstructorDescriptor =>
+          typeAliasDesc.getUnderlyingConstructorDescriptor
+        case _ =>
+          desc
+      }
+    val fullName = descFullNameInternal(dealiasedDesc).map(_.reverse.mkString(""))
+    fullName
+  }
+
+  def funcDescSignature(functionDesc: FunctionDescriptor): Option[String] = {
+    val originalDesc = functionDesc.getOriginal
+
+    val extRecvDesc         = Option(originalDesc.getExtensionReceiverParameter)
+    val extRecvTypeFullName = extRecvDesc.flatMap(paramDesc => typeFullName(paramDesc.getType))
+
+    if (extRecvDesc.nonEmpty && extRecvTypeFullName.isEmpty) {
+      return None
+    }
+
+    val paramTypeFullNames = originalDesc.getValueParameters.asScala.map(paramDesc => typeFullName(paramDesc.getType))
+    if (paramTypeFullNames.exists(_.isEmpty)) {
+      return None
+    }
+
+    val returnTypeFullName =
+      if (isConstructorDesc(originalDesc)) {
+        Some("void")
+      } else {
+        typeFullName(originalDesc.getReturnType)
+      }
+    if (returnTypeFullName.isEmpty) {
+      return None
+    }
+
+    val combinedParamTypeFn = paramTypeFullNames.prepended(extRecvTypeFullName)
+
+    val signature = s"${returnTypeFullName.get}(${combinedParamTypeFn.flatten.mkString(",")})"
+    Some(signature)
+  }
+
+  def combineFunctionFullName(descFullName: String, signature: String): String = {
+    s"$descFullName:$signature"
+  }
+
+  def typeFullName(typ: KotlinType): Option[String] = {
+    val kotlinFullName =
+      typ.getConstructor.getDeclarationDescriptor match {
+        case classDesc: ClassDescriptor =>
+          descFullName(classDesc)
+        case typeParamDesc: TypeParameterDescriptor =>
+          val upperBoundTypeFns = typeParamDesc.getUpperBounds.asScala.map(typeFullName)
+          if (upperBoundTypeFns.exists(_.isEmpty)) {
+            None
+          } else {
+            Some(upperBoundTypeFns.flatten.mkString("&"))
+          }
+      }
+
+    kotlinFullName.map { kotlinFullName =>
+      val javaFullName = builtinTypeTranslationTable.get(kotlinFullName)
+      if (javaFullName.isDefined) {
+        javaFullName.get
+      } else if (kotlinFullName.startsWith("kotlin.Function")) {
+        // For historical reasons kotlin.Function types are not mapped via JavaToKotlinClassMap.
+        // In order to not touch the policies, we mimic that for now.
```

Comment:
I think we should make the switch. We don't have too many such policies yet, and we can write a simple script to duplicate them during the migration.

Reply:
Done.